cfme/tests/cloud_infra_common/test_relationships.py

```diff
@@ -407,9 +407,12 @@ def test_change_network_security_groups_per_page_items(setup_provider, appliance
     """
     view = navigate_to(provider, "NetworkSecurityGroup")
     view.toolbar.view_selector.select('List View')
-    for item_count in [10, 20, 50]:
-        view.paginator.set_items_per_page(item_count)
-        assert len(view.entities.get_all()) <= view.paginator.items_per_page
+    view.paginator.set_items_per_page(1000)
+    total_count = len(view.entities.get_all())
+    for item_count in [5, 10, 20, 50, 100, 200, 500, 1000]:
+        if item_count <= total_count:
+            view.paginator.set_items_per_page(item_count)
+            assert len(view.entities.get_all()) <= view.paginator.items_per_page
 
```

Comment:
Can we add ```else: break```? It will come out of the for loop once ```if``` condition fails otherwise for loop will iterate for each value.

Reply:
updated